src/Microsoft.Build.Tasks.Git.UnitTests/GitRepositoryTests.cs

```diff
@@ -344,17 +344,26 @@ public void GetSubmoduleHeadCommitSha()
             var gitDir = temp.CreateDirectory();
             var workingDir = temp.CreateDirectory();
 
-            var submoduleGitDir = temp.CreateDirectory();
+            var basicSubmoduleGitDir = temp.CreateDirectory();
 
-            var submoduleWorkingDir = workingDir.CreateDirectory("sub").CreateDirectory("abc");
-            submoduleWorkingDir.CreateFile(".git").WriteAllText("gitdir: " + submoduleGitDir.Path + "\t \v\f\r\n\n\r");
+            var basicSubmoduleWorkingDir = workingDir.CreateDirectory("sub").CreateDirectory("abc");
+            basicSubmoduleWorkingDir.CreateFile(".git").WriteAllText("gitdir: " + basicSubmoduleGitDir.Path + "\t \v\f\r\n\n\r");
 
-            var submoduleRefsHeadsDir = submoduleGitDir.CreateDirectory("refs").CreateDirectory("heads");
-            submoduleRefsHeadsDir.CreateFile("master").WriteAllText("0000000000000000000000000000000000000000");
-            submoduleGitDir.CreateFile("HEAD").WriteAllText("ref: refs/heads/master");
+            var basicSubmoduleRefsHeadsDir = basicSubmoduleGitDir.CreateDirectory("refs").CreateDirectory("heads");
+            basicSubmoduleRefsHeadsDir.CreateFile("master").WriteAllText("0000000000000000000000000000000000000000");
+            basicSubmoduleGitDir.CreateFile("HEAD").WriteAllText("ref: refs/heads/master");
+
+            // this is a unusual but legal case which can occur with older versions of Git or other tools.
+            // see https://git-scm.com/docs/gitsubmodules#_forms for more details.
+            var oldStyleSubmoduleWorkingDir = workingDir.CreateDirectory("old-style-submodule");
+            var oldStyleSubmoduleGitDir = oldStyleSubmoduleWorkingDir.CreateDirectory(".git");
+            var oldStyleSubmoduleRefsHeadDir = oldStyleSubmoduleGitDir.CreateDirectory("refs").CreateDirectory("heads");
+            oldStyleSubmoduleRefsHeadDir.CreateFile("branch1").WriteAllText("1111111111111111111111111111111111111111");
+            oldStyleSubmoduleGitDir.CreateFile("HEAD").WriteAllText("ref: refs/heads/branch1");
```

Comment:
I'd prefer this case to be a separate test to keep the tests simpler.

Reply:
Split this out into a separate test.